lib/pescarte/modulo_pesquisa/relatorio_compiler.ex

```diff
@@ -0,0 +1,41 @@
+defmodule Pescarte.RelatorioCompiler do
+  alias Pescarte.ModuloPesquisa.Repository
+  alias PescarteWeb.RelatorioHTML
+
+  def compilar_relatorios(relatorios_selecionados) do
+    relatorios = Enum.map(relatorios_selecionados, &find_relatorio/1)
+    htmls = Enum.map(relatorios, &RelatorioHTML.content/1)
+    pdfs = Enum.reduce(htmls, [], fn html, acc -> [{:html, html} | acc] end)
+
+    with {:ok, pdf_compilado} <- ChromicPDF.print_to_pdf(pdfs),
+         zip_file <- criar_zip_file(pdf_compilado) do
+      elem(zip_file, 1)
+    end
+  end
```

Comment:
ok essa é um pouco mais complexa porque no pico serão 63 relatórios com em torno de até 10mb o que seria mais 500mb em memória… por isso optei por criar um arquivo temporário com o `ChromicPDF.print_to_pdf/2` e passar um “callback” do controller pra dentro desse módulo, enviando assim o download e não tendo que manter o arquivo em memória durante todo o ciclo. (nosso servidor n é parrudo, n aguentaria tudo isso em RAM)

```suggestion
  def compilar_relatorios(relatorios_selecionados, continuation) do
    with {:ok, htmls} <- parse_relatorios_html(relatorios_selecionados) do
      ChromicPDF.print_to_pdf(htmls, output: fn tmp_path ->
        with {:ok, zip_filename, zip_file} <- criar_zip_file(pdf_compilado) do
          continuation.(zip_filename, zip_file)
        end 
      end)
    end
  end
  
  defp parse_relatorios_html(relatorios_ids) do
    Enum.reduce_while(relatorios_ids, [], fn relatorio_id, htmls ->
      if relatorio = Repository.fetch_relatorio_pesquisa_by_id(relatorio_id) do
        html = RelatorioHTML.content(relatorio)
        {:cont, [{:html, html} | htmls]}
      else
        {:halt, {:error, :pdf_not_found}}
      end
    end)
  end
```